vermouth/geometry.py

```diff
@@ -36,43 +36,56 @@ def angle(vectorBA, vectorBC):
     denominator = np.linalg.norm(vectorBA) * np.linalg.norm(vectorBC)
     cosine = nominator / denominator
     # Floating errors at the limits may cause issues.
-    if cosine > 1:
-        cosine = 1
-    elif cosine < -1:
-        cosine = -1
+    cosine = np.clip(cosine, -1, 1)
     return np.arccos(cosine)
 
 
-def dihedral(vectorAB, vectorBC, vectorCD):
+def dihedral(coordinates):
```

Comment:
I liked the previous signature better. But if you argue this is a low-level function, sure. Just make sure there's also human usable ones which calculate torsion angles and angles between planes respectively.

Reply:
In practice I always used all dihedral calculation functions with 4 points. It is what gromacs, MDAnalysis, or VMD do. I could have a 6 points version, or a 2 times 3 point version as well.

Reply:
The problem with a 4pt version is that impropers and torsion angles become separate things. I do agree it's the most common thing you'll find.